test/project_types/node/commands/serve_test.rb

```diff
@@ -76,6 +55,27 @@ def test_update_env_with_host
         run_cmd('app node serve --host="https://example-foo.com"')
       end
 
+      def test_server_command_when_invalid_host_passed
+        invalid_host = "garbage://example.com"
+        ShopifyCLI::Tasks::UpdateDashboardURLS.expects(:call).never
+        ShopifyCLI::Resources::EnvFile.any_instance.expects(:update).never
+        @context.expects(:system).with(
+          "npm run dev",
+          env: {
+            "SHOPIFY_API_KEY" => "mykey",
+            "SHOPIFY_API_SECRET" => "mysecretkey",
+            "SHOP" => "my-test-shop.myshopify.com",
+            "SCOPES" => "read_products",
+            "HOST" => "garbage://example.com",
+            "PORT" => "8081",
+          }
+        ).never
+
+        assert_raises ShopifyCLI::Abort do
+          run_cmd("app node serve --host=#{invalid_host}")
+        end
+      end
+
```

Comment:
I updated the invalid host unit test to take in the invalid host url directly from the command instead of setting `Tunnel.start` (ngrok) since `Tunnel.start` would likely always return a valid host url. This allows the tests to pass with the new changes. @pepibumur @paulomarg